pkg/autoscaler/k8sclient/k8sclient.go

```diff
@@ -140,17 +162,31 @@ type ClusterStatus struct {
 }
 
 func (k *k8sClient) GetClusterStatus() (clusterStatus *ClusterStatus, err error) {
-	opt := metav1.ListOptions{Watch: false}
-
-	nodes, err := k.clientset.CoreV1().Nodes().List(opt)
-	if err != nil || nodes == nil {
+	// TODO: Consider moving this to NewK8sClient method and failing fast when
+	// reflector can't initialize. That is a tradeoff vs silently non-working
+	// component (where in huge majority of cases restarting won't help anyway)
```

Comment:
I'm not sure if we should say 'it won't help in huge majority of cases' because it also may not help to be silently non-working in huge majority of cases (unless we have reason for extra cases there) :)

Reply:
So what I have on mind is that in majority of cases no matter if we restart or not, the end result will be non-working CPA. It just may give better visibility. Rephrasing.